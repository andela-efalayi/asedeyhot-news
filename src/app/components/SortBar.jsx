import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Toolbar, ToolbarGroup } from 'material-ui/Toolbar';
import SelectField from 'material-ui/SelectField';
import MenuItem from 'material-ui/MenuItem';
import { lightBlue500 } from 'material-ui/styles/colors';

const styles = {
  underlineStyle: {
    borderColor: lightBlue500,
  }
};

class SortBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      id: this.props.id
    };
  }
  render() {
    return (
      <Toolbar id={this.props.id} className="toolbar">
        <ToolbarGroup>
          <h4 className="sortBarTitle">{this.props.title}
            <span> - {this.props.currentSortOption}</span>
          </h4>
        </ToolbarGroup>
        <ToolbarGroup>
          <SelectField underlineStyle={styles.underlineStyle}
          floatingLabelText="SortBy Options"
          onChange={this.props.filterHeadlines}
          >
            {this.props.sortOptions.map(option => (
              <MenuItem key={option} value={option} primaryText={option}
              />
            ))}
          </SelectField>
        </ToolbarGroup>
      </Toolbar>
    );
  }
}

SortBar.propTypes = {
  currentSortOption: PropTypes.string,
  id: PropTypes.string,
  title: PropTypes.string,
  sortOptions: PropTypes.array,
  filterHeadlines: PropTypes.func
};

SortBar.defaultProps = {
  currentSortOption: null,
  id: null,
  title: null,
  sortOptions: [],
  filterHeadlines: null
};


export default SortBar;

