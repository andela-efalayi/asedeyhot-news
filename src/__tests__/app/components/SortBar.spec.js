import React from 'react';
import { shallow } from 'enzyme';
import SortBar from '../../../app/components/SortBar.jsx';

jest.dontMock('../../../app/components/SortBar.jsx');

const sortBarDetails = {
  currentSortOption: 'top',
  title: 'ABC News (AU)',
  sortOptions: ['top']
};

describe('SortBar.jsx', () => {
  const wrapper = shallow(<SortBar
    sortOptions={sortBarDetails.sortOptions}
    id={sortBarDetails.id}
    title={sortBarDetails.title} />);
  it('should match component snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should have a select field with one option item', () => {
    expect(wrapper.nodes[0]
  .props.children[1].props.children.props.children.length).toEqual(1);
  });
  it('should have have option item with key "top"', () => {
    expect(wrapper.nodes[0].props.children[1].props
    .children.props.children[0].key).toEqual('top');
  });
});
