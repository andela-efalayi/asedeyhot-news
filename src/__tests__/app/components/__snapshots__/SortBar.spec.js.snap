// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBar.jsx should match SortBar component snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Toolbar
    className="filter"
    noGutter={false}
>
    <ToolbarGroup
        firstChild={false}
        lastChild={false}
    >
        <h4
            className="sortBarTitle"
        >
            ABC News (AU)
            <span>
                 - 
                 News
            </span>
        </h4>
    </ToolbarGroup>
    <ToolbarGroup
        className="select"
        firstChild={false}
        lastChild={false}
    >
        <SelectField
            autoWidth={false}
            disabled={false}
            floatingLabelText="SortBy Options"
            fullWidth={false}
            multiple={false}
            onChange={undefined}
        >
            <MenuItem
                anchorOrigin={
                    Object {
                        "horizontal": "right",
                        "vertical": "top",
                      }
                }
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                primaryText="top"
                value="top"
            />
        </SelectField>
    </ToolbarGroup>
</Toolbar>,
  "nodes": Array [
    <Toolbar
      className="filter"
      noGutter={false}
>
      <ToolbarGroup
            firstChild={false}
            lastChild={false}
      >
            <h4
                  className="sortBarTitle"
            >
                  ABC News (AU)
                  <span>
                         - 
                         News
                  </span>
            </h4>
      </ToolbarGroup>
      <ToolbarGroup
            className="select"
            firstChild={false}
            lastChild={false}
      >
            <SelectField
                  autoWidth={false}
                  disabled={false}
                  floatingLabelText="SortBy Options"
                  fullWidth={false}
                  multiple={false}
                  onChange={undefined}
            >
                  <MenuItem
                        anchorOrigin={
                              Object {
                                    "horizontal": "right",
                                    "vertical": "top",
                                  }
                        }
                        checked={false}
                        desktop={false}
                        disabled={false}
                        focusState="none"
                        insetChildren={false}
                        primaryText="top"
                        value="top"
                  />
            </SelectField>
      </ToolbarGroup>
</Toolbar>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SortBar
        id={undefined}
        sortOptions={
                Array [
                        "top",
                      ]
        }
        title="ABC News (AU)"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SortBar {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "id": undefined,
          "sortOptions": Array [
            "top",
          ],
          "title": "ABC News (AU)",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Toolbar
          className="filter"
          noGutter={false}
>
          <ToolbarGroup
                    firstChild={false}
                    lastChild={false}
          >
                    <h4
                              className="sortBarTitle"
                    >
                              ABC News (AU)
                              <span>
                                         - 
                                         News
                              </span>
                    </h4>
          </ToolbarGroup>
          <ToolbarGroup
                    className="select"
                    firstChild={false}
                    lastChild={false}
          >
                    <SelectField
                              autoWidth={false}
                              disabled={false}
                              floatingLabelText="SortBy Options"
                              fullWidth={false}
                              multiple={false}
                              onChange={undefined}
                    >
                              <MenuItem
                                        anchorOrigin={
                                                  Object {
                                                            "horizontal": "right",
                                                            "vertical": "top",
                                                          }
                                        }
                                        checked={false}
                                        desktop={false}
                                        disabled={false}
                                        focusState="none"
                                        insetChildren={false}
                                        primaryText="top"
                                        value="top"
                              />
                    </SelectField>
          </ToolbarGroup>
</Toolbar>,
        "_debugID": 2,
        "_renderedOutput": <Toolbar
          className="filter"
          noGutter={false}
>
          <ToolbarGroup
                    firstChild={false}
                    lastChild={false}
          >
                    <h4
                              className="sortBarTitle"
                    >
                              ABC News (AU)
                              <span>
                                         - 
                                         News
                              </span>
                    </h4>
          </ToolbarGroup>
          <ToolbarGroup
                    className="select"
                    firstChild={false}
                    lastChild={false}
          >
                    <SelectField
                              autoWidth={false}
                              disabled={false}
                              floatingLabelText="SortBy Options"
                              fullWidth={false}
                              multiple={false}
                              onChange={undefined}
                    >
                              <MenuItem
                                        anchorOrigin={
                                                  Object {
                                                            "horizontal": "right",
                                                            "vertical": "top",
                                                          }
                                        }
                                        checked={false}
                                        desktop={false}
                                        disabled={false}
                                        focusState="none"
                                        insetChildren={false}
                                        primaryText="top"
                                        value="top"
                              />
                    </SelectField>
          </ToolbarGroup>
</Toolbar>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SortBar
    id={undefined}
    sortOptions={
        Array [
            "top",
          ]
    }
    title="ABC News (AU)"
/>,
}
`;
